rxnetty/src/main/java/io/reactivex/netty/pipeline/PrimitiveConversionHandler.java

```diff
@@ -0,0 +1,34 @@
+package io.reactivex.netty.pipeline;
+
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.channel.ChannelHandler.Sharable;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelOutboundHandlerAdapter;
+import io.netty.channel.ChannelPromise;
+
+@Sharable
+public class PrimitiveConversionHandler extends ChannelOutboundHandlerAdapter {
+
+    public static final PrimitiveConversionHandler INSTANCE = new PrimitiveConversionHandler();
+
+    @Override
+    public void write(ChannelHandlerContext ctx, Object msg, ChannelPromise promise) throws Exception {
+        Object msgToWrite = msg;
+
+        if (msg instanceof String) {
```

Comment:
I don't actually know the difference between `ByteBufUtil.writeUtf8()` and `writeBytes(String.getBytes())` but I assume functionally they are equivalent. (assuming the default char encoding is "UTF-8")